src/cache.cc

```diff
@@ -451,14 +451,26 @@ long CACHE::operate()
       champsim::transform_while_n(translation_stash, std::back_inserter(inflight_tag_check), initiate_tag_bw, is_translated, initiate_tag_check<false>());
   initiate_tag_bw.consume(stash_bandwidth_consumed);
   std::vector<long long> channels_bandwidth_consumed{};
+
+  if(std::size(upper_levels) > 1) {
+    std::rotate(upper_levels.begin(), upper_levels.begin() + 1, upper_levels.end());
+  }
+
+  //upper levels get an equal portion of the remaining bandwidth
+  champsim::bandwidth::maximum_type per_upper_bandwidth{(champsim::bandwidth::maximum_type)(std::max((size_t)initiate_tag_bw.amount_remaining() / std::size(upper_levels),size_t{1}))};
```

Comment:
This is where the test is failing. The cache in the test in question has zero upper levels: it only ever issues prefetches. This division has zero in the denominator.

Reply:
I guess what we set it to really shouldn't matter.

Reply:
Right, the particulars don't matter because we won't do anything with the value, we should just protect the division against the zero case.

Reply:
```
  champsim::bandwidth::maximum_type per_upper_bandwidth = std::size(upper_levels) >= 1 ?
                                                          champsim::bandwidth::maximum_type{(champsim::bandwidth::maximum_type)(std::max((size_t)initiate_tag_bw.amount_remaining() / std::size(upper_levels),size_t{1}))} :
                                                          champsim::bandwidth::maximum_type{};
```

Reply:
Man, that syntax sure is clunky. It was made intentionally clunky (so that you don't accidentally xor it with an address, for example), but it's clunky.

Reply:
I applied the suggested patch (less some redundancy).